flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTable.java

```diff
@@ -412,7 +441,7 @@ public boolean tryMarkSlotActive(JobID jobId, AllocationID allocationId) {
 	public boolean isSlotFree(int index) {
 		TaskSlot taskSlot = taskSlots.get(index);
 
-		return taskSlot.isFree();
+		return taskSlot == null || taskSlot.isFree();
```

Comment:
True, we could consider removing `TaskSlotState#FREE`, but why `ACTIVE`?

Reply:
See above. I'll remove `TaskSlotState#FREE` but keep `ACTIVE` as discussed offline. I'll also remove `TaskSlot#isFree` because there should never be a free task slot anymore.